example/lib/main.dart

```diff
@@ -42,88 +52,93 @@ class _MyHomePageState extends State<MyHomePage> {
       })
       ..onConnection(handler: ({required uhstSocket}) {
         uhstSocket
           ..onDiagnostic(handler: ({required message}) {
-            setState(() {
-              hostMessages.add(message);
-            });
+            hostMessages.add(message);
+            setState(() {});
           })
           ..onMessage(handler: ({required message}) {
-            setState(() {
-              hostMessages
-                  .add('Host received: $message from ${uhstSocket.remoteId}');
-            });
+            hostMessages
+                .add('Host received: $message from ${uhstSocket.remoteId}');
+            setState(() {});
             host?.broadcastString(message: message);
           })
           ..onOpen(handler: () {
-            setState(() {
-              hostMessages.add('Client ${uhstSocket.remoteId} connected');
-            });
+            hostMessages.add('Client ${uhstSocket.remoteId} connected');
+            setState(() {});
+          })
+          ..onClose(handler: ({required hostId}) {
+            hostMessages.add('Client ${uhstSocket.remoteId} disconected');
+            setState(() {});
           });
       });
   }
 
+  Future<void> disconnectHost() async => host?.disconnect();
   void initUHST() {
-    uhst ??= UHST(debug: true);
+    uhst ??= UHST(
+      debug: true,
+      relayUrl: 'http://127.0.0.1:3000',
+    );
   }
 
-  Future<void> join() async {
+  Future<void> joinHost() async {
     initUHST();
     client?.close();
     client = uhst?.join(hostId: _hostIdController.text);
     client
       ?..onException(handler: ({required dynamic exception}) {
         if (exception is InvalidHostId || exception is InvalidClientOrHostId) {
-          setState(() {
-            clientMessages.add('Invalid hostId!');
-          });
+          clientMessages.add('Invalid hostId!');
         } else {
-          setState(() {
-            clientMessages.add(exception.toString());
-          });
+          clientMessages.add(exception.toString());
         }
+        setState(() {});
       })
       ..onDiagnostic(handler: ({required message}) {
-        setState(() {
-          clientMessages.add(message);
-        });
+        clientMessages.add(message);
+        setState(() {});
       })
       ..onOpen(handler: () {
-        setState(() {
-          clientMessages.add('Client connected to host: ${client?.remoteId}');
-        });
+        clientMessages.add('Client connected to host: ${client?.remoteId}');
+        setState(() {});
+      })
+      ..onClose(handler: ({required hostId}) {
+        clientMessages.add('Connection to host $hostId dropped.');
+        setState(() {});
       })
       ..onMessage(handler: ({required message}) {
-        setState(() {
-          clientMessages.add('Client received: $message');
-        });
+        clientMessages.add('Client received: $message');
+        setState(() {});
       });
   }
 
+  Future<void> leaveHost() async => client?.close();
```

Comment:
maybe better name can be "disconnect" instead of "leaveHost"

Reply:
fixed